bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/type/ModuleTypeRegistryImpl.java

```diff
@@ -73,22 +76,23 @@ public ModuleType get(String typeUID) {
     }
 
     @Override
-    public <T extends ModuleType> Collection<T> getByTag(String moduleTypeTag) {
+    public <T extends ModuleType> Collection<T> getByTag(@Nullable String moduleTypeTag) {
         return getByTag(moduleTypeTag, null);
     }
 
     @Override
     @SuppressWarnings("unchecked")
-    public <T extends ModuleType> Collection<T> getByTag(String moduleTypeTag, Locale locale) {
+    public <T extends ModuleType> Collection<T> getByTag(@Nullable String moduleTypeTag, @Nullable Locale locale) {
         Collection<T> result = new ArrayList<>(20);
         forEach((provider, mType) -> {
             ModuleType mt = locale == null ? mType
                     : ((ModuleTypeProvider) provider).getModuleType(mType.getUID(), locale);
-            Collection<String> tags = mt.getTags();
-            if (moduleTypeTag == null) {
-                result.add((T) createCopy(mt));
-            } else if (tags.contains(moduleTypeTag)) {
-                result.add((T) createCopy(mt));
+            if ((mt != null) && (moduleTypeTag == null || mt.getTags().contains(moduleTypeTag))) {
```

Comment:
```suggestion
            if (mt != null && (moduleTypeTag == null || mt.getTags().contains(moduleTypeTag))) {
```

Reply:
On other projects my Eclipse is configured to automatically remove redundant parentheses as save cleanup action. :wink: